app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,69 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+  
+  def show
+    task_id = params[:id]
+    @task = Task.find_by(id: task_id)
+    if @task.nil?
+      redirect_to root_path
+      return
+    end
+  end
+  
+  def new
+    @task = Task.new
+  end
+  
+  def create
+    @task = Task.new(
+      name: params[:task][:name],
+      description: params[:task][:description],
+      created_at: params[:task][:created_at])
+      if @task.save
+        redirect_to task_path(@task.id)
+        return
+      else
+        render :new, :bad_request
```

Comment:
I think this works, but I like to do this being more explicit about the status code.

```suggestion
        render :new, status: :bad_request
```